app/validators/questionnaire_validator.py

```diff
@@ -134,3 +140,115 @@ def validate_introduction_block(self):
         )
         if not has_introduction_blocks:
             self.add_error(error_messages.PREVIEW_WITHOUT_INTRODUCTION_BLOCK)
+
+    def validate_answer_references(self):
+
+        # Handling blocks in group
+        for group in self.questionnaire_schema.groups:
+            self.validate_answer_source_group(group)
+
+        # Handling section level "enabled" rule
+        for index, section in enumerate(self.questionnaire_schema.sections):
+            self.validate_answer_source_section(section, index)
+
+    def validate_answer_source_group(self, group):
+        identifier_references = get_object_containing_key(group, "source")
+        for path, identifier_reference, parent_block in identifier_references:
+            # set up default parent_block_id for later check (group or block level)
+            parent_block_id = None
+            if (
+                "source" in identifier_reference
+                and identifier_reference["source"] == "answers"
+            ):
+
+                source_block = self.questionnaire_schema.get_block_by_answer_id(
+                    identifier_reference["identifier"]
+                )
+                # Handling non-existing blocks used as source
+                if not source_block:
+                    self.add_error(
+                        ValueSourceValidator.ANSWER_SOURCE_REFERENCE_INVALID,
+                        identifier=identifier_reference["identifier"],
+                    )
+                    return False
```

Comment:
I had a learning question as I'm trying to understand the eQ schema layouts - Is there a reason why it doesn't continue the loop here, and instead returns on the first encounter of `ANSWER_SOURCE_REFERENCE_INVALID`?

Reply:
Yeah, no worries, the `return` finishes the whole validation execution with a negative result (for a given schema), the `add_error` method only adds errors to the list. I copied it from another place, where a `True` or `False` output is expected, but I guess here we can continue the validation so I can remove the return.

Reply:
Cool that makes sense, thanks

Reply:
In this case we need to exit as the source block won't exist when later we try to access (access `source_block["type"]` it in:
```python
def resolve_source_block_id(self, source_block):
        # Handling of source block nested (list collector's add-block)
       if source_block["type"] == "ListAddQuestion":
```
so we need to exit early

Reply:
Ahh I see, okay cool, thanks!